src/core_modules/capture-core/components/Pages/Search/SearchPage.component.js

```diff
@@ -80,126 +65,184 @@ const getStyles = (theme: Theme) => ({
+                              { programId, programName, searchGroups },
+                          ],
+
+                      },
+                  };
+              }, {}),
+      [],
+      );
+
+    const searchOptions: AvailableSearchOptions =
+      useMemo(() =>
+          Object.values(trackedEntityTypesWithCorrelatedPrograms)
+              // $FlowFixMe https://github.com/facebook/flow/issues/2221
+              .reduce((acc, { trackedEntityTypeId, trackedEntityTypeName, trackedEntityTypeSearchGroups, programs }) =>
+                  ({
+                      ...acc,
+                      [trackedEntityTypeId]:
+                         buildSearchOption(
+                             trackedEntityTypeId,
+                             trackedEntityTypeName,
+                             trackedEntityTypeSearchGroups,
+                             searchScopes.TRACKED_ENTITY_TYPE,
+                         ),
+                      ...programs.reduce((acc2, { programId, programName, searchGroups }) => ({
+                          ...acc2,
+                          [programId]: buildSearchOption(programId, programName, searchGroups, searchScopes.PROGRAM),
+                      }), {}),
+                  }), {}),
+      [trackedEntityTypesWithCorrelatedPrograms],
+      );
+
+    const searchStatus: string =
+      useSelector(({ searchPage }) => searchPage.searchStatus, isEqual);
+
+    const generalPurposeErrorMessage: string =
+      useSelector(({ searchPage }) => searchPage.generalPurposeErrorMessage, isEqual);
+
+    const preselectedProgram: SelectedSearchScope =
+      useSelector(({ currentSelections }) => {
+          const preselected = Object.values(trackedEntityTypesWithCorrelatedPrograms)
+              // $FlowFixMe https://github.com/facebook/flow/issues/2221
+              .map(({ programs }) => programs.find(({ programId }) => programId === currentSelections.programId))
+              .filter(program => program)[0];
+          return {
+              value: preselected && preselected.programId,
+              label: preselected && preselected.programName,
+          };
+      }, isEqual);
 
-const Index = ({
-    addFormIdToReduxStore,
-    navigateToMainPage,
-    showInitialSearchPage,
-    classes,
-    trackedEntityTypesWithCorrelatedPrograms,
-    preselectedProgram,
-    availableSearchOptions,
-    searchStatus,
-    generalPurposeErrorMessage,
-}: Props) => {
     const [selectedSearchScope, setSelectedSearchScope] = useState(preselectedProgram);
 
-    const handleSearchScopeSelection = (program) => {
-        showInitialSearchPage();
-        setSelectedSearchScope(program);
-    };
+    const searchGroupForSelectedScope =
+      (selectedSearchScope.value ? searchOptions[selectedSearchScope.value].searchGroups : []);
 
     useEffect(() => {
         if (!preselectedProgram.value) {
-            showInitialSearchPage();
+            dispatchShowInitialSearchPage();
```

Comment:
dunno what you think about that. I kinda like the fact that we can understand this hides a dispatch function by its name. The data-visualizer-app does something similar. Let me know what you think tho :)

Reply:
Isnt the component still knowing the same thing? I mean the function and its dependencies is the same. The only this that changes is that the dev knows what the function does without having to look further into it.

Reply:
https://github.com/dhis2/capture-app/pull/870#discussion_r487989971

Reply:
@JoakimSM I have applied the changes